on-premise-docker-registry/README.md

```diff
@@ -0,0 +1,191 @@
+
+This sample shows a simple configuration with nginx as a reverse proxy. You can use any other reverse proxy implementation based on your on-premise Docker registry behavior and APIs.
+
+- Create namespace and enable istio sidecar injection if not done earlier.
+
+```shell
+kubectl create namespace ${NAMESPACE}
+kubectl label namespace ${NAMESPACE} istio-injection=enabled
+```
+
+- Deploy the nginx as a reverse proxy. The following components will be deployed:
+  - [ConfigMap](./k8s/configmap.yaml) for nginx configuration
+  - [Deployment and Service](./k8s/deployment.yaml)
+  - [PeerAuthentication](./k8s/peer-authentication.yaml) set to `PERMISSIVE` to allow for communication between kubelet and the nginx reverse proxy.
+
+```shell
+make deploy-nginx-reverse-proxy
+```
+
+- Wait for the nginx reverse proxy to be up and running
+
+```shell
+make check-nginx-reverse-proxy
+```
+
+- Export NodePort for the nginx reverse proxy as an environment variable
+
+```shell
+export NGINX_NODE_PORT=$(kubectl get svc nginx -o jsonpath='{.spec.ports[0].nodePort}')
+```
+
+## Test workload
+
+Let's deploy a sample workload that will use an image from the on-premise Docker registry.
+
+- Log in to the created on-premise docker registry
+
+```shell
+make docker-login
+```
+
+- Create and store a sample Docker image in the on-premise Docker registry. It uses an image tag based on the current time.
+
+```shell
+make create-test-image
+```
+
+- Update the [deployment.yaml for test workload](./test-image-deployment/deployment.yaml).
+  - Replace `{nginx-reverse-proxy-node-port}` with the `NGINX_NODE_PORT` value.
+  - Replace `{generate-image-tag}` with the tag of the generated image.
+
+- Create Kubernetes Secret with credentials to pull the Docker image.
+
+```shell
+make create-secret-to-pull-image
+```
+
+- Deploy the test workload. Notice that in the [deployment file](./test-image-deployment/deployment.yaml), the image is specified as `localhost:30930/....`. This is the localhost wrt to the Kubernetes worker node, and the port is the NodePort of the nginx reverse proxy service.
+
+```shell
+make deploy-test-workload
+```
+
+- Wait until the Pod is up and running. Check the status using the following command:
+
+```shell
+make check-test-workload
+```
+
+- Once the Pod is up and running, you can access the application
+
+```shell
+make access-test-workload
+```
+
+## Cleanup
+
+- Remove the test workload, the nginx reverse proxy, and the local Docker registry
+
+```shell
+make cleanup
+```
+
+- Remove the entry from Cloud Connector.
+
+## Takeaways
+
+- It is possible to have a setup to pull docker images from an on-premise docker registry for applications deployed on SAP BTP Kyma runtime
+- Docker images are pulled using connectivity proxy and Cloud Connector.
+- A reverse proxy component would need to be installed from the customer side. This will adapt and forward the requests to on-premise docker registry via connectivity proxy.
```

Comment:
```suggestion
- The reverse proxy component must be installed from the customer side. This will adapt and forward the requests to the on-premise Docker registry using connectivity proxy.
```